### PR TITLE
doc: fix typos in config.rst

### PR DESCRIPTION
**Pull request by shun-s**
Fix several typos in config.rst

Signed-off-by: Song Shun <song.shun3@zte.com.cn>

### CODE REVIEW

File: doc/dev/config.rst [hunk 1]
```diff
@@ -93,22 +93,22 @@ calling these functions to make sure your changes get applied.
 Defining config options
 =======================
 
-New-style config options are defined in common/options.cc.  All new config
-options should go here (and not into legacy_config_opts.h).
+New-style config options are defined in common/options.cc. All new config
+options should go here (and not into legacy config_opts.h).
```
Comment (joscollin) [inline doc/dev/config.rst]: This seems to be wrong. There is a file src/common/legacy_config_opts.h

Comment (shun-s) [inline doc/dev/config.rst]: yeah, that's true. 
already recovery to legacy_config_opts.h.
thanks

**Comment by joscollin** [comment]
Jenkins retest this please

**Comment by joscollin** [comment]
Jenkins retest this please

**Comment by joscollin** [comment]
Jenkins retest this please

File: doc/dev/config.rst [hunk 2]
```diff
@@ -57,7 +57,7 @@ Reading configuration values
 There are two ways for Ceph code to get configuration values. One way is to
 read it directly from a variable named "g_conf," or equivalently,
 "g_ceph_ctx->_conf." The other is to register an observer that will be called
-every time the relevant configuration values changes.  This observer will be
+every time the relevant configuration values changes. This observer will be
```
Comment (tchaikov) [inline doc/dev/config.rst]: @shun-s @joscollin why is this considered a typo? is this an accepted guide line to have a single space to start a new sentence in our document? or is it recommended by some best practice ? if yes, for what reason?

for more counter examples, see https://github.com/ceph/ceph/pull/16611/files

**Comment by joscollin** [comment]
@ZVampirEM77 Before merging, I have verified that the change made for `legacy_config_opts.h` doesn't appear in the `Files Changed` tab any more and reverted. But for some reason the wrong text was merged. I will create a new PR now to fix it. Thanks for the information.

File: doc/dev/config.rst (hunk 2 above)
Comment (joscollin) [inline doc/dev/config.rst]: @tchaikov I found that these minor changes looks better along with other typos that are fixed by @shun-s in this PR. The mistake I did was: somehow I missed the  `legacy_config_opts.h` revert. Created a new PR for that: https://github.com/ceph/ceph/pull/16721.

Comment (tchaikov) [inline doc/dev/config.rst]: @joscollin my questions still hold.

Comment (tchaikov) [inline doc/dev/config.rst]: and i'd suggest avoid these sort of "minor changes" if possible unless they are piggy-backed by more significant changes. IMO, they are not typo at all. and there are not visible change introduced by this commit. please try to build the script and check the output.

Comment (joscollin) [inline doc/dev/config.rst]: @tchaikov Yes, I completely accept your suggestion. But there were other significant changes in this PR along with these minors.

Comment (tchaikov) [inline doc/dev/config.rst]: like the comma removals?

Comment (joscollin) [inline doc/dev/config.rst]: @tchaikov Yes, they were unnecessary. Looks like a copy-paste mistake with those options.

**Pull request merged**